File: Java/src/test/java/org/chess/rendering/ConsoleRendererTest.java
package org.chess.rendering;

import org.chess.domain.GameState;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;

class ConsoleRendererTest {

    @Test
    void shouldRenderEmptyBoard() {
        final var renderer = new ConsoleRenderer();
        final var gameState = new GameState(Map.of());

        var actual = renderer.render(gameState);
        var expected = """
                                   
                     ╔═════╦═════╦═════╦═════╦═════╦═════╦═════╦═════╗
                  8  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  7  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  6  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  5  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  4  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  3  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  2  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  1  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╚═════╩═════╩═════╩═════╩═════╩═════╩═════╩═════╝
                        a     b     c     d     e     f     g     h
                """;

        assertThat(actual).isEqualTo(expected);
    }

    @Test
    void shouldRenderInitialPositions() {
        final var renderer = new ConsoleRenderer();
        final var gameState = GameState.initialState();

        var actual = renderer.render(gameState);
        var expected = """
                                   
                     ╔═════╦═════╦═════╦═════╦═════╦═════╦═════╦═════╗
                  8  ║  ♜  ║  ♞  ║  ♝  ║  ♛  ║  ♚  ║  ♝  ║  ♞  ║  ♜  ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  7  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  6  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  5  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  4  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  3  ║     ║     ║     ║     ║     ║     ║     ║     ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  2  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║
                     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
                  1  ║  ♖  ║  ♘  ║  ♗  ║  ♕  ║  ♔  ║  ♗  ║  ♘  ║  ♖  ║
                     ╚═════╩═════╩═════╩═════╩═════╩═════╩═════╩═════╝
                        a     b     c     d     e     f     g     h
                """;

        assertThat(actual).isEqualTo(expected);
    }
}

File: Kotlin/src/test/kotlin/org/chess/rendering/ConsoleRendererTest.kt
package org.chess.rendering

import org.assertj.core.api.Assertions
import org.chess.domain.GameState
import org.junit.jupiter.api.Test

internal class ConsoleRendererTest {

    @Test
    fun shouldRenderEmptyBoard() {
        val renderer = ConsoleRenderer()
        val gameState = GameState(emptyMap())
        val actual = renderer.render(gameState)
        val expected = """
     ╔═════╦═════╦═════╦═════╦═════╦═════╦═════╦═════╗
  8  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  7  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  6  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  5  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  4  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  3  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  2  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  1  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╚═════╩═════╩═════╩═════╩═════╩═════╩═════╩═════╝
        a     b     c     d     e     f     g     h
"""

        Assertions.assertThat(actual).isEqualTo(expected)
    }

    @Test
    fun shouldRenderInitialPositions() {
        val renderer = ConsoleRenderer()
        val gameState = GameState.initialState()
        val actual = renderer.render(gameState)
        val expected = """
     ╔═════╦═════╦═════╦═════╦═════╦═════╦═════╦═════╗
  8  ║  ♜  ║  ♞  ║  ♝  ║  ♛  ║  ♚  ║  ♝  ║  ♞  ║  ♜  ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  7  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║  ♟  ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  6  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  5  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  4  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  3  ║     ║     ║     ║     ║     ║     ║     ║     ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  2  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║  ♙  ║
     ╠═════╬═════╬═════╬═════╬═════╬═════╬═════╬═════╣
  1  ║  ♖  ║  ♘  ║  ♗  ║  ♕  ║  ♔  ║  ♗  ║  ♘  ║  ♖  ║
     ╚═════╩═════╩═════╩═════╩═════╩═════╩═════╩═════╝
        a     b     c     d     e     f     g     h
"""

        Assertions.assertThat(actual).isEqualTo(expected)
    }
}
